cmd/dp-import-api/main.go

```diff
@@ -11,147 +11,145 @@ import (
-	if err != nil {
-		log.ErrorC("direct kafka producer error", err, nil)
-		os.Exit(1)
-	}
-
-	router := mux.NewRouter()
+	var serviceList initialise.ExternalServiceList
 
-	healthcheckHandler := healthcheck.NewMiddleware(handlershealthcheck.Handler)
-	identityHandler := identity.Handler(cfg.ZebedeeURL)
+	mongoDataStore, err := serviceList.GetMongoDataStore(cfg)
+	logIfError(err, "mongodb datastore error")
 
-	// TODO how long should the ID be?
-	alice := alice.New(requestID.Handler(16), healthcheckHandler, identityHandler).Then(router)
+	dataBakerProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.DatabakerImportTopic, initialise.DataBaker, cfg.KafkaMaxBytes)
+	logIfError(err, "databaker kafka producer error")
 
-	httpServer := server.New(cfg.BindAddr, alice)
-	httpServer.HandleOSSignals = false
+	directProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.InputFileAvailableTopic, initialise.Direct, cfg.KafkaMaxBytes)
+	logIfError(err, "direct kafka producer error")
 
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	auditProducer, err := serviceList.GetProducer(cfg.Brokers, cfg.AuditEventsTopic, initialise.Audit, cfg.KafkaMaxBytes)
+	logIfError(err, "direct kafka producer error")
 
 	urlBuilder := url.NewBuilder(cfg.Host, cfg.DatasetAPIURL)
 	jobQueue := importqueue.CreateImportQueue(dataBakerProducer.Output(), directProducer.Output())
 
-	datasetAPI := dataset.API{client, cfg.DatasetAPIURL, cfg.ServiceAuthToken}
-	recipeAPI := recipe.API{client, cfg.RecipeAPIURL}
+	client := rchttp.NewClient()
+	datasetAPI := dataset.API{Client: client, URL: cfg.DatasetAPIURL, ServiceAuthToken: cfg.ServiceAuthToken}
+	recipeAPI := recipe.API{Client: client, URL: cfg.RecipeAPIURL}
 
 	jobService := job.NewService(mongoDataStore, jobQueue, &datasetAPI, &recipeAPI, urlBuilder)
 	auditor := audit.New(auditProducer, serviceNamespace)
 
-	_ = api.CreateImportAPI(router, mongoDataStore, jobService, auditor)
+	api.CreateImportAPI(cfg.Host, cfg.BindAddr, cfg.ZebedeeURL, mongoDataStore, jobService, auditor)
 
-	// signals the web server shutdown, so a graceful exit is required
-	httpErrChannel := make(chan error)
-	// launch web server in background
 	go func() {
-		log.Debug("listening...", log.Data{"bind_address": cfg.BindAddr})
-		if err := httpServer.ListenAndServe(); err != nil {
-			log.Error(err, nil)
-			httpErrChannel <- err
-			return
+		var databaKerProducerErrors, directProducerErrors, auditProducerErrors chan (error)
+
+		if serviceList.DataBakerProducer {
+			databaKerProducerErrors = dataBakerProducer.Errors()
+		} else {
+			databaKerProducerErrors = make(chan error, 1)
+		}
+
+		if serviceList.DirectProducer {
+			directProducerErrors = directProducer.Errors()
+		} else {
+			directProducerErrors = make(chan error, 1)
+		}
+
+		if serviceList.AuditProducer {
+			auditProducerErrors = auditProducer.Errors()
+		} else {
+			auditProducerErrors = make(chan error, 1)
+		}
+
+		select {
```

Comment:
As discussed we need to revisit this in the future - as this will only ever deal with the first error. Once the first error has been logged the go routine will complete and the app will block until the OS sig is recieved.